mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -307,7 +307,13 @@ def _get_gpu_type_and_quantity(resources):
 
     def _validate(self, runobj: RunObject):
         # validating length limit for sparkjob's function name
-        verify_field_regex("run.metadata.name", runobj.metadata.name, sparkjob_name)
+        try:
+            verify_field_regex("run.metadata.name", runobj.metadata.name, sparkjob_name)
+
+        except mlrun.errors.MLRunInvalidArgumentError:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Job name '{runobj.metadata.name}' is not valid. The job name must be not longer than 29 characters"
```

Comment:
You're making a decision here that the validation failed on name length, but if you'll look on the validation for sparkjob_name you'll see that for example the validation may failed cause you gave invalid characters `/` or the first/last character is `-`/`_`/`.`
Ideally we should have an error message per regex validation so you won't have to do this wrapping everywhere, something like [k8s do](https://github.com/kubernetes/kubernetes/blob/v1.20.0/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L156), obviously this is something more infrastructure-y in MLRun and out of context for your PR (we will add it to our backlog), but, I do think that you should align your error message to be more exact on what the job name must be, or practically, change it to be similar to the one in the k8s link I sent above

Reply:
In addition please do raise X from Y so we won't lose the stack trace: